tests/test_context_managers.py

```diff
@@ -24,18 +29,30 @@ def setUp(self):
 
     def tearDown(self):
         self.conn.close()
-    
+
     def test_context(self):
         with self.conn.root.context(3) as x:
             print( "entering test" )
             self.assertTrue(on_context_enter)
+            self.assertFalse(on_context_exc)
             self.assertFalse(on_context_exit)
             print( "got past context enter" )
             self.assertEqual(x, 20)
             print( "got past x=20" )
+        self.assertFalse(on_context_exc)
         self.assertTrue(on_context_exit)
         print( "got past on_context_exit" )
 
+    def test_context_exception(self):
+        with self.assertRaises(AssertionError):
```

Comment:
There is two things wrong here:
- having multiple asserts within the `with` block means we don't know which one generated the exception
- checking against `AssertionError` confuses the boundaries between test failures and desired behaviour

I think the test should look like this:

```
with self.conn.root.context(3):
    raise ValueError()
self.assertTrue(on_context_enter)
self.assertTrue(on_context_error)
self.assertTrue(on_context_exit)
```

(without the `print`, don't ask me why they are in the other test..)

and remove the `raise` line from the context manager.

Reply:
To clarify, is this PR a fix for locally raising an exception within the context of a remote context manager, or a fix for remotely reraising it from within the `__exit__` handler of the context manager?

If the first one is the case, this test should be simplified as I suggested.